setup.py

```diff
@@ -42,7 +42,7 @@
         'jinja2',
         'nc-time-axis',  # needed by iris.plot
         'nested-lookup',
-        'netCDF4',
+        'netCDF4!=1.6.1',  # github.com/ESMValGroup/ESMValCore/issues/1723
```

Comment:
These changes look unrelated to this pull request?

Reply:
eagle-eyed Bouwe :eagle: but also "Man, you weren't there!" (Vietnam reference) - check the whole panic from yesterday when @schlunma and myself were trying to figure out why the tests are dying spectacularly with a flurry of SegFaults - also this https://github.com/Unidata/netcdf4-python/issues/1192 and this https://github.com/ESMValGroup/ESMValCore/pull/1724

Reply:
BTW remember our old SegFaults that eventually almost totally died out like Covid? Back with the vengeance :laughing:

Reply:
Awesome detective work you both. Since #1724 has now been merged, could you merge `main` into this branch, so the diff shows only the changes you actually want to make here?

Reply:
done, bud :beer: